t/cobrand/oxfordshire.t

```diff
@@ -183,6 +184,47 @@ FixMyStreet::override_config {
         };
     }
 
+    subtest 'extra data sent with defect update' => sub {
+        my $comment = FixMyStreet::DB->resultset('Comment')->first;
+        $comment->set_extra_metadata(defect_raised => 1);
+        $comment->update;
+        $comment->problem->external_id('hey');
+        $comment->problem->update;
+
+        my $cbr = Test::MockModule->new('FixMyStreet::Cobrand::Oxfordshire');
+        $cbr->mock('_fetch_features', sub { [ {
+            type => 'Feature',
+            geometry => { type => 'LineString', coordinates => [ [ 1, 2 ], [ 3, 4 ] ] },
+            properties => { USRN => 12345, STREET => 'Street Avenue', TOWN_NAME => 'Town' },
+        } ] });
+        my $test_res = HTTP::Response->new();
+        $test_res->code(200);
+        $test_res->message('OK');
+        $test_res->content('<?xml version="1.0" encoding="utf-8"?><service_request_updates><request_update><update_id>248</update_id></request_update></service_request_updates>');
+
+        my $o = Open311->new(
+            fixmystreet_body => $oxon,
+            test_mode => 1,
+            test_get_returns => { 'servicerequestupdates.xml' => $test_res },
+        );
+
+        $o->post_service_request_update($comment);
+        my $cgi = CGI::Simple->new($o->test_req_used->content);
+        is $cgi->param('attribute[usrn]'), 12345, 'USRN sent with update';
+        is $cgi->param('attribute[raise_defect]'), 1, 'Defect flag sent with update';
+        is $cgi->param('attribute[street_descriptor]'), '12345 - Street Avenue, Town', 'Street description sent';
+
+        # Now set a USRN on the problem (found at submission)
+        $comment->problem->push_extra_fields({ name => 'usrn', value => '12345' });
+        $comment->problem->update;
+
+        $o->post_service_request_update($comment);
+        $cgi = CGI::Simple->new($o->test_req_used->content);
+        is $cgi->param('attribute[usrn]'), 12345, 'USRN sent with update';
+        is $cgi->param('attribute[raise_defect]'), 1, 'Defect flag sent with update';
```

Comment:
Given that the results of these two tests are the same, how would we know if one wasn't working?

Reply:
I'm not sure what you mean, sorry - if the report already has a USRN (from the JS if it's the road/pavement categories) then it looks up only that USRN for the extra data. If it doesn't, then it looks up the closest USRN and its data. In both cases, the result should be the same, in that it has success. If you mean the mock is the same in both, I didn't think that mattered to the test, but could have a different mock if you think it does. Or let me know what I've misunderstood :)

Reply:
Yeah, it's me being picky about the mock using the same data. I guess I think it should be possible for the test to fail if it doesn't use the USRN which isn't the case here.